src/wrappers/themis/jsthemis/secure_session.cpp

```diff
@@ -146,6 +146,13 @@ void SecureSession::New(const Nan::FunctionCallbackInfo<v8::Value>& args)
             return;
         }
         SecureSession* obj = new SecureSession(id, private_key, v8::Local<v8::Function>::Cast(args[2]));
+        if (!obj->isCreated()) {
+            // secure_session_create() inside SecureSession::SecureSession()
+            // may fail, we need to catch this and throw exception to JS
+            ThrowParameterError("SecureSession", "unsupported private key");
+            args.GetReturnValue().SetUndefined();
```

Comment:
Initially I wanted to note that there may be other reasons for a failure here, but most of them are actually checked and reported before this call. It can still fail due to allocation errors and such but that's rare so I think this message should be fine.

Reply:
Unfortunately, `secure_session_create()` only gives `NULL` or some valid pointer, and we can't get the actual error code in case of error.